doc/advanced-configuration/host_config.md

```diff
@@ -0,0 +1,229 @@
+The `host_config` section is used to configure options for specific XMPP domains.
+For each domain requiring such options, a `host_config` section needs to be created with the following format:
+
+* **Scope:** for each option the scope is the same as for the corresponding top-level option.
+* **Syntax:** domain subsection starts with `[[host_config]]` and contains the options listed below.
+* **Default:** none - all domain-level options need to be specified explicitly.
+* **Example:** see the examples for each section below.
+
+**Note:** Each hosted domain needs to be included in the list of [`hosts`](general.md#generalhosts) in the `general` section.
+
+# General options
+
+### `host_config.host`
+
+* **Syntax:** string, domain name
+* **Default:** no default, this option is mandatory
+* **Example:** `host = "my-xmpp-server.com"`
+
+This option specifies the XMPP domain that this section refers to.
+
+# Configuration sections
+
+The following sections are accepted in `host_config`:
+
+## `host_config.general`
+
+The options defined here override the ones defined in the top-level [`general`](general.md) section.
+The following options are allowed:
+
+* [`pgsql_users_number_estimate`](general.md#generalpgsql_users_number_estimate)
+* [`route_subdomains`](general.md#generalroute_subdomains)
+* [`replaced_wait_timeout`](general.md#generalreplaced_wait_timeout)
+* [`hide_service_name`](general.md#generalhide_service_name)
+
+#### Example
+
+The `hide_service_name` option is set to `false` only for `domain2.com`.
+
+```toml
+[general]
+  hosts = ["domain1.com", "domain2.com", "domain3.com"]
+  loglevel = "info"
+  hide_service_name = true
+  replaced_wait_timeout = 1000
+
+[[host_config]]
+  host = "domain2.com"
+
+  [host_config.general]
+    hide_service_name = false
+```
+
+## `host_config.auth`
+
+This section completely overrides the top-level [`auth`](auth.md) section. All options are allowed.
+
+#### Example
+
+The intention here is to enable only the `SASL PLAIN` mechanism for `domain2.com`.
+To do this, we need to specify `methods` as well - otherwise there would be no authentication methods
+enabled for `domain2.com` as the entire `auth` section is replaced.
+
+```toml
+[general]
+  hosts = ["domain1.com", "domain2.com", "domain3.com"]
+
+[auth]
+  methods = ["internal"]
+
+[[host_config]]
+  host = "domain2.com"
+
+  [host_config.auth]
+    methods = ["internal"]
+    sasl_mechanisms = ["plain"]
+```
+
+## `host_config.modules`
+
+This section completely overrides the top-level [`modules`](modules.md) section. All options are allowed.
+
+#### Example
+
+The modules enabled for `domain2.com` will be `mod_disco` and `mod_stream_management`.
+If we wanted to enable `mod_roster`, it would need to be repeated in `host_config`.
```

Comment:
Hmmm... that's interesting. Wouldn't we expect enabling mod_roster globally to be inherited by all the hosts, unless explicitly stated otherwise? Not sure which approach is better, so I'm looking for some thoughts on this 🤔

Reply:
That's, how it works now. One tuple `{modules, ...}` = all or nothing, no way to inherit module by module.